spring-web/src/main/java/org/springframework/http/client/JdkClientHttpRequest.java

```diff
@@ -48,8 +51,19 @@ class JdkClientHttpRequest extends AbstractStreamingClientHttpRequest {
 	 * The JDK HttpRequest doesn't allow all headers to be set. The named headers are taken from the default
 	 * implementation for HttpRequest.
 	 */
-	private static final List<String> DISALLOWED_HEADERS =
-			List.of("connection", "content-length", "expect", "host", "upgrade");
+	protected static final Set<String> DISALLOWED_HEADERS = getDisallowedHeaders();
```

Comment:
`protected` was changed to test without much setup.

Reply:
`AbstractMockWebServerTests` dispatch for `/echo` checks that `Host` is `localhost:port`, so unsure how to test otherwise.